numpyro/distributions/constraints.py

```diff
@@ -432,6 +433,39 @@ def log_abs_det_jacobian(self, x, y, intermediates=None):
         return x[..., -n:].sum(-1)
 
 
+class MultivariateAffineTransform(Transform):
+    r"""
+    Transform via the mapping :math:`y = loc + scale\_tril\ @\ x`.
+
+    :param loc: a real vector.
+    :param scale_tril: a lower triangular matrix with positive diagonal.
+    """
+    domain = real_vector
+    codomain = real_vector
+    event_dim = 1
+
+    def __init__(self, loc, scale_tril):
+        # TODO: relax this condition per user request
+        if np.ndim(scale_tril) != 2:
+            raise ValueError("Only support 2-dimensional scale_tril matrix.")
+        self.loc = loc
+        self.scale_tril = scale_tril
+
+    def __call__(self, x):
+        return self.loc + np.squeeze(np.matmul(self.scale_tril, x[..., np.newaxis]), axis=-1)
```

Comment:
nit: why not just `self.scale_tril @ x`?

Reply:
`x` here is a vector, so we have to use matrix-vector product here. We can replace matmul by @ but we still need to unsqueeze and squeeze.

Reply:
I think @ should support matrix-vector product:

```
In [3]: np.ones((4, 3)) @ np.ones(3)
Out[3]: DeviceArray([3., 3., 3., 3.], dtype=float32)
```

Reply:
But it will fail when `x` has batch dimension, so we can't compose this transform with distributions having non-trivial batch_shape.

Reply:
Oh, now I understand. Just for my own notes, this will fail on:

```python
>>> np.ones((4, 3)) @ np.ones([2, 3])
TypeError: dot_general requires contracting dimensions to have the same shape, got [3] and [2].

>>> np.matmul(np.ones((4, 3)), np.ones([2, 3])[..., np.newaxis])
DeviceArray([[[3.],
              [3.],
              [3.],
              [3.]],

             [[3.],
              [3.],
              [3.],
              [3.]]], dtype=float32)

```